IllegalStateException on `onPartitionsAssigned`: Cannot finish block 6074 when current block is 6074

Seen in the soak env. 2 consecutive log events:

```
2026-03-13T07:44:21.258355711Z ERROR o.a.k.c.c.i.ConsumerRebalanceListenerInvoker
[Consumer clientId=consumer-xtdb-xtdb-log-4, groupId=xtdb-xtdb-log] User provided listener xtdb.api.log.KafkaCluster$KafkaLog$openGroupConsumer$1$1 failed on invocation of onPartitionsAssigned for partitions [xtdb-log-0]
java.lang.IllegalStateException: Cannot finish block 6074 when current block is 6074
	at xtdb.catalog.BlockCatalog.buildBlock(BlockCatalog.kt:68)
	at xtdb.indexer.BlockFinisher.finishBlock(BlockFinisher.kt:70)
	at xtdb.indexer.LogProcessor.onPartitionsAssigned(LogProcessor.kt:123)
	at xtdb.api.log.KafkaCluster$KafkaLog$openGroupConsumer$1$1.onPartitionsAssigned(KafkaCluster.kt:362)
	at org.apache.kafka.clients.consumer.internals.ConsumerRebalanceListenerInvoker.invokePartitionsAssigned(ConsumerRebalanceListenerInvoker.java:64)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.onJoinComplete(ConsumerCoordinator.java:464)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:510)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:421)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:550)
	at org.apache.kafka.clients.consumer.internals.ClassicKafkaConsumer.updateAssignmentMetadataIfNeeded(ClassicKafkaConsumer.java:680)
	at org.apache.kafka.clients.consumer.internals.ClassicKafkaConsumer.poll(ClassicKafkaConsumer.java:644)
	at org.apache.kafka.clients.consumer.internals.ClassicKafkaConsumer.poll(ClassicKafkaConsumer.java:625)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:895)
	at xtdb.api.log.KafkaCluster$KafkaLog$PollingConsumer$pollingJob$1$1$2$1.invoke(KafkaCluster.kt:307)
	at xtdb.api.log.KafkaCluster$KafkaLog$PollingConsumer$pollingJob$1$1$2$1.invoke(KafkaCluster.kt:305)
	at kotlinx.coroutines.InterruptibleKt.runInterruptibleInExpectedContext(Interruptible.kt:48)
	at kotlinx.coroutines.InterruptibleKt.access$runInterruptibleInExpectedContext(Interruptible.kt:1)
	at kotlinx.coroutines.InterruptibleKt$runInterruptible$2.invokeSuspend(Interruptible.kt:40)
	at kotlinx.coroutines.InterruptibleKt$runInterruptible$2.invoke(Interruptible.kt)
	at kotlinx.coroutines.InterruptibleKt$runInterruptible$2.invoke(Interruptible.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:66)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:43)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:157)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at kotlinx.coroutines.InterruptibleKt.runInterruptible(Interruptible.kt:39)
	at kotlinx.coroutines.InterruptibleKt.runInterruptible$default(Interruptible.kt:36)
	at xtdb.api.log.KafkaCluster$KafkaLog$PollingConsumer$pollingJob$1$1$2.invokeSuspend(KafkaCluster.kt:305)
	at xtdb.api.log.KafkaCluster$KafkaLog$PollingConsumer$pollingJob$1$1$2.invoke(KafkaCluster.kt)
	at xtdb.api.log.KafkaCluster$KafkaLog$PollingConsumer$pollingJob$1$1$2.invoke(KafkaCluster.kt)
	at kotlinx.coroutines.selects.SelectImplementation$ClauseData.invokeBlock(Select.kt:843)
	at kotlinx.coroutines.selects.SelectImplementation.complete(Select.kt:715)
	at kotlinx.coroutines.selects.SelectImplementation.doSelect$suspendImpl(Select.kt:444)
	at kotlinx.coroutines.selects.SelectImplementation.doSelect(Select.kt)
	at xtdb.api.log.KafkaCluster$KafkaLog$PollingConsumer$pollingJob$1.invokeSuspend(KafkaCluster.kt:479)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
```

```
2026-03-13T07:44:21.266684573Z ERROR xtdb.util
Uncaught exception:
org.apache.kafka.common.KafkaException: User rebalance callback throws an error
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.onJoinComplete(ConsumerCoordinator.java:470)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:510)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:421)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:550)
	at org.apache.kafka.clients.consumer.internals.ClassicKafkaConsumer.updateAssignmentMetadataIfNeeded(ClassicKafkaConsumer.java:680)
	at org.apache.kafka.clients.consumer.internals.ClassicKafkaConsumer.poll(ClassicKafkaConsumer.java:644)
	at org.apache.kafka.clients.consumer.internals.ClassicKafkaConsumer.poll(ClassicKafkaConsumer.java:625)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:895)
	at xtdb.api.log.KafkaCluster$KafkaLog$PollingConsumer$pollingJob$1$1$2$1.invoke(KafkaCluster.kt:307)
	at xtdb.api.log.KafkaCluster$KafkaLog$PollingConsumer$pollingJob$1$1$2$1.invoke(KafkaCluster.kt:305)
	at kotlinx.coroutines.InterruptibleKt.runInterruptibleInExpectedContext(Interruptible.kt:48)
	at kotlinx.coroutines.InterruptibleKt.access$runInterruptibleInExpectedContext(Interruptible.kt:1)
	at kotlinx.coroutines.InterruptibleKt$runInterruptible$2.invokeSuspend(Interruptible.kt:40)
	at kotlinx.coroutines.InterruptibleKt$runInterruptible$2.invoke(Interruptible.kt)
	at kotlinx.coroutines.InterruptibleKt$runInterruptible$2.invoke(Interruptible.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:66)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:43)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:157)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at kotlinx.coroutines.InterruptibleKt.runInterruptible(Interruptible.kt:39)
	at kotlinx.coroutines.InterruptibleKt.runInterruptible$default(Interruptible.kt:36)
	at xtdb.api.log.KafkaCluster$KafkaLog$PollingConsumer$pollingJob$1$1$2.invokeSuspend(KafkaCluster.kt:305)
	at xtdb.api.log.KafkaCluster$KafkaLog$PollingConsumer$pollingJob$1$1$2.invoke(KafkaCluster.kt)
	at xtdb.api.log.KafkaCluster$KafkaLog$PollingConsumer$pollingJob$1$1$2.invoke(KafkaCluster.kt)
	at kotlinx.coroutines.selects.SelectImplementation$ClauseData.invokeBlock(Select.kt:843)
	at kotlinx.coroutines.selects.SelectImplementation.complete(Select.kt:715)
	at kotlinx.coroutines.selects.SelectImplementation.doSelect$suspendImpl(Select.kt:444)
	at kotlinx.coroutines.selects.SelectImplementation.doSelect(Select.kt)
	at xtdb.api.log.KafkaCluster$KafkaLog$PollingConsumer$pollingJob$1.invokeSuspend(KafkaCluster.kt:479)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: null
Caused by: java.lang.IllegalStateException: Cannot finish block 6074 when current block is 6074
	at xtdb.catalog.BlockCatalog.buildBlock(BlockCatalog.kt:68)
	at xtdb.indexer.BlockFinisher.finishBlock(BlockFinisher.kt:70)
	at xtdb.indexer.LogProcessor.onPartitionsAssigned(LogProcessor.kt:123)
	at xtdb.api.log.KafkaCluster$KafkaLog$openGroupConsumer$1$1.onPartitionsAssigned(KafkaCluster.kt:362)
	at org.apache.kafka.clients.consumer.internals.ConsumerRebalanceListenerInvoker.invokePartitionsAssigned(ConsumerRebalanceListenerInvoker.java:64)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.onJoinComplete(ConsumerCoordinator.java:464)
	... 36 common frames omitted
```

--- a/core/src/main/kotlin/xtdb/indexer/LogProcessor.kt
+++ b/core/src/main/kotlin/xtdb/indexer/LogProcessor.kt
@@ -112,9 +112,9 @@ class LogProcessor(
                     LOG.debug("transition: replica watchers caught up to $replayTarget")
 
                     val followerProc = oldSys.proc
-                    val pendingBlock = followerProc.pendingBlock
-                    LOG.debug("transition: closing follower system (pendingBlock=${pendingBlock != null})")
+                    LOG.debug("transition: closing follower system")
                     oldSys.close()
+                    val pendingBlock = followerProc.pendingBlock
 
                     procFactory.openTransition(replicaProducer, replicaWatchers).use { transition ->
                         if (pendingBlock != null) {

